exoctk/contam_visibility/sossFieldSim.py

```diff
@@ -232,6 +236,237 @@ def sossFieldSim(ra, dec, binComp='', dimX=256):
+    for kPA in range(PAtab.size):
+        APA = PAtab[kPA]
+        V3PA = APA+0.57  # from APT
+        sindx = np.sin(np.pi/2+APA/radeg)*stars['dDEC']
+        cosdx = np.cos(np.pi/2+APA/radeg)*stars['dDEC']
+        ps = pixel_scale
+        stars['dx'] = (np.cos(np.pi/2+APA/radeg)*stars['dRA']-sindx)/ps
+        stars['dy'] = (np.sin(np.pi/2+APA/radeg)*stars['dRA']+cosdx)/ps
+        stars['x'] = stars['dx']+sweetSpot['x']
+        stars['y'] = stars['dy']+sweetSpot['y']
+
+        # Display the star field (blue), target (red), subarray (green),
+        # full array (blue), and axes
+        if (kPA == 0 and nStars > 1) and False:
+            print(kPA)
+            plt.plot([0, 2047, 2047, 0, 0], [0, 0, 2047, 2047, 0], 'b')
+            plt.plot([0, 255, 255, 0, 0], [0, 0, 2047, 2047, 0], 'g')
+            # the order 1 & 2 traces
+            path = '/Users/david/Documents/work/jwst/niriss/soss/data/'
+            t1 = np.loadtxt(path+'trace_order1.txt', unpack=True)
+            plt.plot(t1[0], t1[1], 'r')
+            t2 = np.loadtxt(path+'trace_order2.txt', unpack=True)
+            plt.plot(t2[0], t2[1], 'r')
+            plt.plot(stars['x'], stars['y'], 'b*')
+            plt.plot(sweetSpot['x'], sweetSpot['y'], 'r*')
+            plt.title("APA= {} (V3PA={})".format(APA, V3PA))
+            ax = plt.gca()
+
+            # add V2 & V3 axes
+            l, hw, hl = 250, 50, 50
+            adx, ady = -l*np.cos(-0.57 / radeg), -l*np.sin(-0.57 / radeg)
+            ax.arrow(2500, 1800, adx, ady, head_width=hw, head_length=hl,
+                     length_includes_head=True, fc='k')  # V3
+            plt.text(2500+1.4*adx, 1800+1.4*ady, "V3", va='center',
+                     ha='center')
+            adx, ady = -l*np.cos((-0.57-90)/radeg), -l*np.sin((-0.57-90)/radeg)
+            ax.arrow(2500, 1800, adx, ady, head_width=hw, head_length=hl,
+                     length_includes_head=True, fc='k')  # V2
+            plt.text(2500+1.4*adx, 1800+1.4*ady, "V2", va='center',
+                     ha='center')
+            # add North and East
+            adx, ady = -l*np.cos(APA/radeg), -l*np.sin(APA/radeg)
+            ax.arrow(2500, 1300, adx, ady, head_width=hw, head_length=hl,
+                     length_includes_head=True, fc='k')  # N
+            plt.text(2500+1.4*adx, 1300+1.4*ady, "N", va='center', ha='center')
+            adx, ady = -l*np.cos((APA-90)/radeg), -l*np.sin((APA-90)/radeg)
+            ax.arrow(2500, 1300, adx, ady, head_width=hw, head_length=hl,
+                     length_includes_head=True, fc='k')  # E
+            plt.text(2500+1.4*adx, 1300+1.4*ady, "E", va='center', ha='center')
+
+            ax.set_xlim(-400, 2047+800)
+            ax.set_ylim(-400, 2047+400)
+            ax.set_aspect('equal')
+            plt.show()
+
+        # Retain stars that are within the Direct Image NIRISS POM FOV
+        ind, = np.where((stars['x'] >= -162) & (stars['x'] <= 2047+185) &

```

Comment:
@exowanderer @hover2pi hi, I'm currently modifying the framework. This is basically the original sossFieldSim() function, but with a new `instrument` parameter, and some small tweaks. Q: Are the numbers `-162` and `-154` arbitrary, or is there a particular reason for these values? I'm wondering if I should change these for NIRCam + MIRI modes.